### DISCUSSION TITLE
Migrating to NextAuth from an existing project

### DISCUSSION THREAD

**Discussion by ThePaulMcBride** [nextauthjs/next-auth#1187] [Help]
I have an existing project where the auth is powered by Firebase. Currently I support sign in with email and allow oAuth sign ins with Github and Google.

When a user signs in for the first time, I copy their details in to my database using prisma, so I have their email address.

I would love to move to NextAuth but need users that previously signed in with oAuth to be able to. Right now while testing I can see that is going to be an issue as when I try to sign in with oAuth I can see the error "To confirm your identity, sign in with the same account you used originally.".

That makes sense as a user account already exists in the database with a matching email address and NextAuth does not support automatic account merging.

What are my options here? Any help would be greatly appreciated!

**Comment by ThePaulMcBride** [comment]
I've done some digging around and looking through other issues and it looks like as I still have access to Firebase I'll be able to import the accounts of current users. I've got this working locally I think. Are there any pitfalls I should be aware of with this approach?

**Reply by jessycormier** [reply]
How did you manage with this issue? Did you find a solution or stumble across any issues?
